l10n_br_account/models/account_move_line.py

```diff
@@ -216,15 +216,40 @@ def create(self, vals_list):
                     )
                 )
         self._inject_shadowed_fields(vals_list)
-        results = super(
+
+        # This reordering is crucial to ensure accurate linkage between
+        # account.move.line (aml) and the fiscal document line. In the fiscal create a
+        # fiscal document line, leaving only those that should be created. Proper
+        # ordering is essential as mismatches between the order of amls and the
+        # manipulated vals_list of fiscal documents can lead to incorrect linkages.
+        # For example, if vals_list[0] in amls does not match vals_list[0] in the
+        # fiscal document (which is a manipulated vals_list), it results in erroneous
+        # associations.
+        if len(vals_list) > 1 and self.env.company.country_id.code == "BR":
```

Comment:
Não olhei profundamente mas essa condição de forma genérica  self.env.company.country_id.code == "BR" não parece ser a melhor opção para identificar os casos que não tem Documentos Fiscais, o melhor parece ser verificar se o account.move tem um fiscal_document_id ou o fiscal_operation_id preenchidos, para atender os casos onde a empresa é do Brasil mas emite tanto Documentos Fiscais quanto Faturas Sem Doc Fiscal/Invoices(account.move original) ou o caso de múltiplas localizações instaladas no mesmo banco de dados

Reply:
verdade e pode correr o risco de estar manipulando a fatura de um outra empresa mas sob o contexto da empresa x..

Reply:
eu coloquei verificação do pais porque havia dado erro em um teste do account que se nao me engano parte do nome e return tag_tax_ids ... talvez com a reordenação do result para o a sequencial original nao seja necessário verificar o pais para resequenciar antes da criação

Reply:
valeu @mbcosta. Vai ser mais robusto sim, vou amendar ja.

Reply:
Amendei. Optei por não adicionar mais teste para ver se o caso é do Brasil, já que a re-ordenaçao é pela presencia do fiscal_operation_line_id, se for empresa de fora não vai alterar a ordenação original...

Outra coisa, @marcelsavegnago eu botei porque `isinstance(vals_list, list)` como teste porque `if len(vals_list) > 1` da True se vals_list for um dict (o que é permitido no create...).

Reply:
ok, mas então isso é melhor melhorar num outro PR. Pois aqui o foco é apenas resolver a regressão de #3064.
Se pilhar para mandar PR...

Reply:
pessoal, eu tirei então os testes para ver se a empresa era do Brasil e deu treta nos testes internacionais aqui:
https://github.com/OCA/l10n-brazil/actions/runs/9069191266/job/24918186228?pr=3077#step:8:2523

Vale a pena a gente analisar, pode ser que algum problema importante foi detetado...

Reply:
@marcelsavegnago @mbcosta @gabrielcardoso21 então deixando o codigo rodar sem procurar se a empresa é do Brasil (porque não é codigo pesado, nem vale nem a complexidade de 2 if), pelo menos descobri algo errado com o sort initial do @marcelsavegnago `key=lambda x: x[1].get("fiscal_operation_line_id") is False` pois podia ter dict que vinha com `"fiscal_operation_line_id": False` e nesse caso uma linha dessa ia na frente da lista. Por isso que um teste international falhava, mas podia acontecer num caso do Brasil tb. Então na nova ordenação eu usei:
`key=lambda x: not x[1].get("fiscal_operation_line_id")` que funciona sempre.

Eu botei isso naquele fixup aqui https://github.com/OCA/l10n-brazil/pull/3077/commits/49c69d32f03ce0231c98be482c0f71c923bbd868
vcs conseguem revisar de novo?

Novamente devido a regressão introduzida pelo PR #3064, esse fix é bem urgente...

cc @renatonlima @antoniospneto